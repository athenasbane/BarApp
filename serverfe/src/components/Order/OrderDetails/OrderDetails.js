import React from 'react';
import {
    Grid,
    Button,
    makeStyles
} from '@material-ui/core';
import OrderItem from '../OrderItem/OrderItem';

const useStyles = makeStyles({
    root: {
        width: '100%'
    },
    buttonBox: {
        marginTop: '10px'
    },
})

const OrderDetails = props => {
    const classes = useStyles();
    const orderItems = props.order.orderedItems.map(item => (
        <OrderItem key={item.optionId} item={item}/>));

    return (
        <Grid className={classes.root} container alignItems="center" justify="center" spacing={5}>
            <Grid item>
                {orderItems}
            </Grid>
            <Grid className={classes.buttonBox} container item direction="row" spacing={2}>
                <Grid item xs={6}>
                    <Button
                        fullWidth
                        onClick={props.toggleColor} variant="contained">Working On</Button>
                </Grid>
                <Grid item xs={6}>
                    <Button 
                        fullWidth 
                        onClick={() => props.deliveredHandler(props.order._id)} 
                        variant="contained">Delivered</Button>
                </Grid>
            </Grid>
        </Grid>
    );
}

export default OrderDetails;