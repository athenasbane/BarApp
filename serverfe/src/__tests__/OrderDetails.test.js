import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import OrderDetails from '../components/Order/OrderDetails/OrderDetails';

const order = {
    createdAt: "2020-10-08T15:02:57.731Z",
    delivered: false,
    orderedItems: [
        {
            optionId: "5f6a3d7bf2723782c641bb35",
            subOption: "Half-Pint",
            title: "Strongbow",
            volume: 3,
            _id: "5f7f2a2112516b19cd01ef61",
        }
    ],
    tableNumber: 1,
    totalPrice: 6,
    updatedAt: "2020-10-08T15:02:57.731Z",
    __v: 0,
    _id: "5f7f2a2112516b19cd01ef60",
};

describe('Order Details Tests', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(<OrderDetails order={order} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

    test('clicking Working On button calls event once', () => {
        const mockHandler = jest.fn();
        const component = render(
            <OrderDetails order={order} toggleColor={mockHandler} deliveredHandler={mockHandler} />
        );

        const workingOnButton = component.getByText('Working On');
        fireEvent.click(workingOnButton);

        expect(mockHandler.mock.calls).toHaveLength(1);

        const deliveredButton = component.getByText('Delivered');
        fireEvent.click(deliveredButton);

        expect(mockHandler.mock.calls).toHaveLength(2);
        
    })
})