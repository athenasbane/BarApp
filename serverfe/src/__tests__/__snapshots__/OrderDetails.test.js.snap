// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Details Tests renders correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiGrid-root makeStyles-root-3 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <strong>
            Title:
          </strong>
           
          Strongbow
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <strong>
            Option:
          </strong>
           
          Half-Pint
        </p>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <strong>
            Volume:
          </strong>
           
          3
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-buttonBox-2 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Working On
        </span>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Delivered
        </span>
      </button>
    </div>
  </div>
</div>
`;
